$OpenBSD: patch-src_krb5-auth-dialog_c,v 1.2 2009/07/16 20:14:20 ajacoutot Exp $

XXX these should be handled by configure checks.
  krb5_get_init_creds_opt_set_pkinit: does not take 11 arguments.
  krb5_get_init_creds_opt_free: the context argument is an MIT extension.

--- src/krb5-auth-dialog.c.orig	Thu Jul  2 15:04:35 2009
+++ src/krb5-auth-dialog.c	Wed Jul 15 14:34:40 2009
@@ -25,6 +25,7 @@
 #include <stdlib.h>
 #include <time.h>
 #include <krb5.h>
+#include <com_err.h>
 #include <stdio.h>
 #include <sys/wait.h>
 #include <string.h>
@@ -483,8 +484,6 @@ ka_auth_heimdal_pkinit(KaApplet* applet, krb5_creds* c
 						    kprincipal,
 						    pk_userid,
 						    pkinit_anchors,
-						    NULL,
-						    NULL,
 						    0,	  /* pk_use_enc_key */
 						    auth_dialog_prompter,
 						    applet, /* data */
@@ -498,7 +497,7 @@ ka_auth_heimdal_pkinit(KaApplet* applet, krb5_creds* c
 		                              0, NULL, opts);
 out:
 	if (opts)
-		krb5_get_init_creds_opt_free(kcontext, opts);
+		krb5_get_init_creds_opt_free(opts);
 	return retval;
 }
 #endif /* ! ENABLE_PKINIT */
@@ -520,7 +519,7 @@ ka_auth_password(KaApplet* applet, krb5_creds* creds,
 					      0, NULL, opts);
 out:
 	if (opts)
-		krb5_get_init_creds_opt_free(kcontext, opts);
+		krb5_get_init_creds_opt_free(opts);
 	return retval;
 }
 
